addons/account_asset/account_asset.py

```diff
@@ -1,351 +1,325 @@
+    method_number = fields.Integer(string='Number of Depreciations', readonly=True, states={'draft': [('readonly', False)]}, default=5, help="The number of depreciations needed to depreciate your asset")
+    method_period = fields.Integer(string='Number of Months in a Period', required=True, readonly=True, default=12, states={'draft': [('readonly', False)]}, help="The amount of time between two depreciations, in months")
+    method_end = fields.Date(string='Ending Date', readonly=True, states={'draft': [('readonly', False)]})
+    method_progress_factor = fields.Float(string='Degressive Factor', readonly=True, default=0.3, states={'draft': [('readonly', False)]})
+    value_residual = fields.Float(compute='_amount_residual', method=True, digits=dp.get_precision('Account'), string='Residual Value')
+    method_time = fields.Selection([('number', 'Number of Depreciations'), ('end', 'Ending Date')], string='Time Method', required=True, readonly=True, default='number', states={'draft': [('readonly', False)]},
+        help="Choose the method to use to compute the dates and number of depreciation lines.\n"
+             "  * Number of Depreciations: Fix the number of depreciation lines and the time between 2 depreciations.\n"
+             "  * Ending Date: Choose the time between 2 depreciations and the date the depreciations won't go beyond.")
+    prorata = fields.Boolean(string='Prorata Temporis', readonly=True, states={'draft': [('readonly', False)]}, help='Indicates that the first depreciation entry for this asset have to be done from the purchase date instead of the first January / Start date of fiscal year')
+    history_ids = fields.One2many('account.asset.history', 'asset_id', string='History', readonly=True)
+    depreciation_line_ids = fields.One2many('account.asset.depreciation.line', 'asset_id', string='Depreciation Lines', readonly=True, states={'draft': [('readonly', False)], 'open': [('readonly', False)]})
+    salvage_value = fields.Float(string='Salvage Value', digits=dp.get_precision('Account'), help="It is the amount you plan to have that you cannot depreciate.", readonly=True, states={'draft': [('readonly', False)]})
+    invoice_id = fields.Many2one('account.invoice', string='Invoice', states={'draft': [('readonly', False)]}, copy=False)
+    type = fields.Selection(related="category_id.type", string='Type', required=True, default='purchase')
+
+    @api.multi
+    def unlink(self):
+        for asset in self:
+            if asset.state in ['open', 'close']:
+                raise UserError(_('You cannot delete a document is in %s state.') % (asset.state,))
+            if asset.account_move_line_ids:
+                raise UserError(_('You cannot delete a document that contains posted lines.'))
+        return super(account_asset_asset, self).unlink()
+
+    @api.model
+    def _get_period(self):
+        periods = self.env['account.period'].find()
         if periods:
             return periods[0]
```

Comment:
hopefully the user is not a women... otherwise it would be weird to ask her to configure her periods